src/function.py

```diff
@@ -68,6 +74,14 @@
 BACKOFF_MULTIPLIER = 2
 # Max length in bytes of the payload
 MAX_PAYLOAD_SIZE = 1000 * 1024
+# Individual request timeout in seconds (non-configurable)
+INDIVIDUAL_REQUEST_TIMEOUT_DURATION = 3
```

Comment:
This seems really short. But shouldn't be greater than the function's configured timeout.

Reply:
The current timeout was 3 seconds **for the whole session** (this is something that I found really weird, provided that the lambda is configured with a timeout of 30 seconds, see the PR description). This is why I chose this value for individual POST requests.

Take a look at the test `test_session_duration_properly_calculated`, which uses the current values to calculate the maximum session timeout. I could increase the individual request timeout up to 8 seconds, this way the overall session timeout would remain at 28 seconds. What do you think?

Reply:
Yeah, I agree. I would use `context.get_remaining_time_in_millis()` at the beginning of an invocation and use that as a max. Then maybe divide remaining time by the number of retries. The backoff doesn't buy you much if we're talking about a 30 second invocation.

Reply:
As discussed offline, we'll go with the current simpler approach and switch to this one if we detect issues.